### DISCUSSION TITLE
Ubuntu22.04 LTSでASP用のcfgが動かない

### DISCUSSION THREAD

**Discussion by Ryohsuke-Takeuchi** [toppers/users-forum#76] [Q&A]
Ubuntu 22.04 LTSのターミナルで、ASPカーネルのビルドをしようとしましたが、
コンフィギュレータ Release 1.9.6(32bit Linux用バイナリ）をダウンロードして、実行モードにしても、ターミナルで実行ファイルとして認められず実行できません。
また、コンフィギュレータ Release 1.9.6ソースをダウンロードしてboostライブラリをインストールしてビルドしても、エラーとなってビルドできません。
最新のLinuxでは、ASP用のコンフィギュレータは、使えないということでしょうか？

**Comment by mitsut** [comment]
新世代カーネルのコンフィギュレータの64bit Linux環境はC++の問題でうまくビルドできません。
そのため次のGitHubリポジトリで修正作業をしています。
https://github.com/mitsut/cfg
64bit Linux向けにビルドしたバイナリも作成していますので、ダウンロードしてお使いいただけます。
https://github.com/mitsut/cfg/releases/tag/1.9.7

Ubuntu22.04でのビルド確認はできていないので、CI環境を作ろうとは思いつつも後回しになっています。
（使っているのは自分くらいだろうと思っていました）

**Comment by h7ga40** [comment]
ASP向けRuby版CFGもあります。
http://dev.toppers.jp/trac_user/contrib/wiki/rubycfg_asp
少し古いので、最新のRubyで動くかは分かりません。
また、動的生成パッケージ向けなので標準パッケージで使うには*.trbも変更する必要があります。
参考まで。

**Reply by hiroakitakada** [reply]
新世代カーネルと第3世代カーネルで静的APIの文法が少し違います。Ruby版CFGを使うと，第3世代カーネルの文法になると思うので，コンフィギュレーションファイルの修正が必要と思います。

**Comment by Ryohsuke-Takeuchi** [comment]
みつさんのcfgコマンドは、22.04 LTSで実行できました。aspのビルドも問題ありません。arm-none-eabi-gccのバージョンは10.3.1ですが、、
cfgのビルド環境も、以下のterminalの通り、ビルドもできました。
![Screenshot from 2022-06-30 17-37-44](https://user-images.githubusercontent.com/91311953/176633241-a9c43c25-6c69-43c0-8f5c-5d91611ca41e.png)
ありがとうございました。

**Reply by mitsut** [reply]
無事ビルドできたようでホッとしました

**Comment by Ryohsuke-Takeuchi** [comment]
みつさんのcfgは、22.04 LTSで、実行も、ビルドもできました。フォーラムに結果を添付しています。 Windows の メール から送信 差出人: Hiroaki TAKADA送信日時: 2022年6月29日 21:58宛先: toppers/users-forumCC: 竹内良輔; Author件名: Re: [toppers/users-forum] Ubuntu22.04 LTSでASP用のcfgが動かない (Discussion #76) 新世代カーネルと第3世代カーネルで静的APIの文法が少し違います。Ruby版CFGを使うと，第3世代カーネルの文法になると思うので，コンフィギュレーションファイルの修正が必要と思います。—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you authored the thread.Message ID: ***@***.***>
